pkg/operator/ceph/config/conditions.go

```diff
@@ -44,46 +45,48 @@ func ConditionExport(context *clusterd.Context, namespace, name string, conditio
+		if err != nil {
+			logger.Errorf("failed to get cluster %v", err)
 		}
-	}
-	conditionMap[newCondition.Type] = newCondition.Status
-	existingCondition := findStatusCondition(*conditions, newCondition.Type)
-	if existingCondition == nil {
-		newCondition.LastTransitionTime = metav1.NewTime(time.Now())
-		newCondition.LastHeartbeatTime = metav1.NewTime(time.Now())
-		*conditions = append(*conditions, newCondition)
-
-	} else if existingCondition.Status != newCondition.Status || existingCondition.Message != newCondition.Message {
-		newCondition.LastTransitionTime = metav1.NewTime(time.Now())
-		existingCondition.Status = newCondition.Status
-		existingCondition.Reason = newCondition.Reason
-		existingCondition.Message = newCondition.Message
-		existingCondition.LastHeartbeatTime = metav1.NewTime(time.Now())
-	}
-	cluster.Status.Conditions = *conditions
-
-	if newCondition.Status == v1.ConditionTrue {
-		cluster.Status.Phase = newCondition.Type
-		if state := translatePhasetoState(newCondition.Type); state != "" {
-			cluster.Status.State = state
+		if conditions == nil {
+			conditions = &cluster.Status.Conditions
+			if cluster.Status.Conditions != nil {
+				conditionMapping(*conditions)
+			}
+		}
+		conditionMap[newCondition.Type] = newCondition.Status
+		existingCondition := findStatusCondition(*conditions, newCondition.Type)
+		if existingCondition == nil {
+			newCondition.LastTransitionTime = metav1.NewTime(time.Now())
+			newCondition.LastHeartbeatTime = metav1.NewTime(time.Now())
+			*conditions = append(*conditions, newCondition)
+
+		} else if existingCondition.Status != newCondition.Status || existingCondition.Message != newCondition.Message {
+			newCondition.LastTransitionTime = metav1.NewTime(time.Now())
+			existingCondition.Status = newCondition.Status
+			existingCondition.Reason = newCondition.Reason
+			existingCondition.Message = newCondition.Message
+			existingCondition.LastHeartbeatTime = metav1.NewTime(time.Now())
+		}
+		cluster.Status.Conditions = *conditions
+
+		if newCondition.Status == v1.ConditionTrue {
+			cluster.Status.Phase = newCondition.Type
+			if state := translatePhasetoState(newCondition.Type); state != "" {
+				cluster.Status.State = state
+			}
+			cluster.Status.Message = newCondition.Message
+			logger.Infof("CephCluster %q status: %q. %q", namespace, cluster.Status.Phase, cluster.Status.Message)
 		}
-		cluster.Status.Message = newCondition.Message
-		logger.Infof("CephCluster %q status: %q. %q", namespace, cluster.Status.Phase, cluster.Status.Message)
-	}
 
-	if _, err := context.RookClientset.CephV1().CephClusters(namespace).Update(cluster); err != nil {
-		logger.Errorf("failed to update cluster condition %v", err)
-	}
-	if newCondition.Type == cephv1.ConditionReady {
-		checkConditionFalse(context, namespace, name)
+		if _, err := context.RookClientset.CephV1().CephClusters(namespace).Update(cluster); err != nil {
+			logger.Errorf("failed to update cluster condition %v", err)
```

Comment:
How about also logging the name of the condition that failed to update

Reply:
Yes. Will do that.